File: ClientApp/src/app/interfaces/transaccion.ts
import { DetalleTransaccion } from "./detalle-transaccion";

export interface Transaccion {
  codigo: number,
  medio_pago?: number,
  nombreMediopago?: string,
  estado?: number,
  total: number,
  fecha?: string,
  concepto?: string,
  identificacionUsuario?: string,
  codigoComercio?: number,
  nombreComercio?: string
}

File: Models/Transaccion.cs
using System;
using System.Collections.Generic;

namespace AppTransacciones.Models
{
    public partial class Transaccion
    {
        public int codigo { get; set; }
        public int? medio_pago { get; set; }
        public int? estado { get; set; }
        public double total { get; set; }
        public string? fecha { get; set; }
        public string? concepto { get; set; }
        public string? identificacionUsuario { get; set; }
        public int? codigoComercio { get; set; }

        public virtual MedioPago? idMedioNavigation { get; set; }
    }
}
